rgw/v2/tests/s3cmd/test_large_object_gc.py

```diff
@@ -0,0 +1,173 @@
+"""
+test_s3cmd - Test large object download with gc using s3cmd
+
+Usage: test_large_object_gc.py -c <input_yaml>
+
+<input_yaml>
+    Note: Following yaml can be used
+    test_large_object_gc.yaml
+
+Operation:
+    Create an user
+    Create a bucket with user credentials
+    Upload large file to bucket
+    Set rgw_gc_obj_min_wait as 5
+    Download uploaded object
+    Verify download is succeeded
+    Delete uploaded object
+    Delete bucket
+"""
+
+
+import argparse
+import json
+import logging
+import os
+import sys
+import time
+import traceback
+
+sys.path.append(os.path.abspath(os.path.join(__file__, "../../../..")))
+
+
+from v2.lib import resource_op
+from v2.lib.admin import UserMgmt
+from v2.lib.exceptions import RGWBaseException, TestExecError
+from v2.lib.rgw_config_opts import CephConfOp, ConfigOpts
+from v2.lib.s3.write_io_info import BasicIOInfoStructure, IOInfoInitialize
+from v2.lib.s3cmd import auth as s3_auth
+from v2.tests.s3_swift import reusable
+from v2.tests.s3cmd import reusable as s3cmd_reusable
+from v2.utils import utils
+from v2.utils.log import configure_logging
+from v2.utils.test_desc import AddTestInfo
+from v2.utils.utils import RGWService
+
+log = logging.getLogger()
+
+
+def test_exec(config):
+    """
+    Executes test based on configuration passed
+    Args:
+        config(object): Test configuration
+    """
+    io_info_initialize = IOInfoInitialize()
+    basic_io_structure = BasicIOInfoStructure()
+    io_info_initialize.initialize(basic_io_structure.initial())
+    umgmt = UserMgmt()
+    ceph_conf = CephConfOp()
+    rgw_service = RGWService()
+    # preparing data
+    user_name = resource_op.create_users(no_of_users_to_create=1)[0]["user_id"]
+    tenant = "tenant"
+    tenant_user_info = umgmt.create_tenant_user(
+        tenant_name=tenant, user_id=user_name, displayname=user_name
+    )
+    umgmt.create_subuser(tenant_name=tenant, user_id=user_name)
+
+    ip_and_port = s3cmd_reusable.get_rgw_ip_and_port()
+    s3_auth.do_auth(tenant_user_info, ip_and_port)
+
+    bucket_name = utils.gen_bucket_name_from_userid(user_name, rand_no=0)
+
+    # Create a bucket
+    s3cmd_reusable.create_bucket(bucket_name)
+    log.info(f"Bucket {bucket_name} created")
+
+    # Upload a 2GB file to bucket
+    uploaded_file_info = s3cmd_reusable.upload_file(bucket_name, file_size=2147483648)
+    uploaded_file = uploaded_file_info["name"]
+    uploaded_file_md5 = uploaded_file_info["md5"]
+    log.info(f"Uploaded file {uploaded_file} to bucket {bucket_name}")
+
+    if config.gc_verification is True:
+        log.info("making changes to ceph.conf")
+        config.rgw_gc_obj_min_wait = 5
+        ceph_conf.set_to_ceph_conf(
+            "global",
+            ConfigOpts.rgw_gc_obj_min_wait,
+            str(config.rgw_gc_obj_min_wait),
+        )
+        log.info("trying to restart services")
+        srv_restarted = rgw_service.restart()
```

Comment:
The `restart` method is incorrect as we check only for the status of `ceph orch restart <service_name>`. It is not a true indicator that the service has restarted as it would only schedule the restart. This is something we would need to fix in a another PR.

Reply:
Sure. Will plan for the separate PR for this.